src/reporter.ts

```diff
@@ -0,0 +1,180 @@
+    this.testItems = new Map();
+    this.promises = [];
+  }
+
+  addRequestToPromisesQueue(promise: any, failMessage: string): void {
+    this.promises.push(promiseErrorHandler(promise, failMessage));
+  }
+
+  finishSuites(): void {
+    this.suites.forEach(({ id }) => {
+      const finishSuiteObj: FinishTestItemObjType = {
+        endTime: this.client.helpers.now(),
+      };
+      const { promise } = this.client.finishTestItem(id, finishSuiteObj);
+      this.addRequestToPromisesQueue(promise, 'Failed to finish suite.');
+    });
+    this.suites.clear();
+  }
+
+  onBegin(): void {
+    const { launch, description, attributes } = this.RPconfig;
+    const startLaunchObj: StartLaunchObjType = {
+      name: launch,
+      startTime: this.client.helpers.now(),
+      attributes,
+      description,
+    };
+    const { tempId, promise } = this.client.startLaunch(startLaunchObj);
+    this.addRequestToPromisesQueue(promise, 'Failed to launch run.');
+    this.launchId = tempId;
+  }
+
+  onTestBegin(test: TestResp): void {
+    //create suite
+    const suiteHasParent = test.parent.parent?._isDescribe;
+    const suiteTitle = suiteHasParent ? test.parent.parent?.title : test.parent.title;
+    if (!this.suites.has(suiteTitle)) {
+      const startSuiteObj: StartTestObjType = {
+        name: suiteTitle,
+        startTime: this.client.helpers.now(),
+        type: TEST_ITEM_TYPES.SUITE,
+      };
+      const suiteObj = this.client.startTestItem(startSuiteObj, this.launchId);
+      this.addRequestToPromisesQueue(suiteObj.promise, 'Failed to start suite.');
+      this.suites.set(suiteTitle, {
+        id: suiteObj.tempId,
+        name: suiteTitle,
+      });
+    }
+    //suite in suite
+    if (suiteHasParent) {
+      if (!this.suites.has(test.parent.title)) {
+        const { id: parentId } = this.suites.get(suiteTitle);
+        const startChildSuiteObj: StartTestObjType = {
+          name: test.parent.title,
+          startTime: this.client.helpers.now(),
+          type: TEST_ITEM_TYPES.TEST,
+        };
+        const suiteObj = this.client.startTestItem(startChildSuiteObj, this.launchId, parentId);
+        this.addRequestToPromisesQueue(suiteObj.promise, 'Failed to start suite.');
+        this.suites.set(test.parent.title, {
+          id: suiteObj.tempId,
+          name: test.parent.title,
+        });
+      }
+    }
+
+    //create steps
+    if (this.suites.get(test.parent.title)) {
+      const { id: parentId } = this.suites.get(test.parent.title);
+      const startTestItem: StartTestObjType = {
+        name: test.title,
+        startTime: this.client.helpers.now(),
+        type: TEST_ITEM_TYPES.STEP,
+      };
+      const stepObj = this.client.startTestItem(startTestItem, this.launchId, parentId);
+      this.addRequestToPromisesQueue(stepObj.promise, 'Failed to start test.');
+      this.testItems.set(test.title, {
+        name: test.title,
+        id: stepObj.tempId,
+      });
+    }
+  }
```

Comment:
Playwright provide `onStepBegin` method. Maybe it will be better place for create steps

Reply:
`onStepBegin` runs for every action, not for `test('', () =>)`
This is not good for us.